docs/int_header/int.lua

```diff
@@ -1,98 +1,151 @@
--- This reflects the INT header prior to the updated documentation
+-- Copyright (c) 2019 Cable Television Laboratories, Inc.
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--    http://www.apache.org/licenses/LICENSE-2.0
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+
+-- This Wireshark plugin reflects the INT header prior to the updated do
+-- cumentation
+
+-- INT protocol example
+-- declare our protocol
+tps_udp_proto = Proto("TPS_INT", "TPS UDP INT Protocol")
+-- create a function to dissect it
+
 
 function octet_to_mac(buff)
-    addr = ""
+    local addr = ""
     for i = 0,5,1
     do
-        octect = buff(i,1):uint()
+        local octect = buff(i, 1):uint()
+        --noinspection StringConcatenationInLoops
         addr = addr  .. string.format("%02X", octect)
         if i < 5
         then
+            --noinspection StringConcatenationInLoops
             addr = addr .. ":"
         end
     end
     return addr
 end
 
--- INT protocol example
--- declare our protocol
-tps_proto = Proto("TPS_INT","TPS INT Protocol")
--- create a function to dissect it
-function tps_proto.dissector(buffer,pinfo,tree)
-    pinfo.cols.protocol = "TPS INT"
+function tps_int_shim(int_tree, shim_buf)
+    -- UDP INT Shim Header - 4 bytes
+    local shim_tree = int_tree:add(shim_buf, "UDP INT Shim Header")
+    shim_tree:add("Type: " .. shim_buf:bitfield(0, 4))
+    shim_tree:add("NPT: " .. shim_buf:bitfield(4, 2))
+    assert(shim_buf:bitfield(6, 2)) -- reserved
```

Comment:
I thought that the assertion check would be asserting that the values are == nil or 0?

Add an error message with each assert.

I think that it should be something like:
local reserved_1 = shim_buf:bitfield(6, 2)
assert(reserved_1 == 0, "Reserved bits 6-8 in UDP INT Shim header are '" + reserved_1 + "' expected '0'")

Reply:
I tried that but lua didn't like it